scripts/e2e-test.sh

```diff
@@ -62,40 +96,92 @@ if [ -z "$DEVICE" ]; then
   exit 1
 fi
 export e2e_pool_device=$DEVICE
+
 if [ -n "$TAG" ]; then
   export e2e_image_tag="$TAG"
 fi
+
 if [ -n "$REGISTRY" ]; then
   export e2e_docker_registry="$REGISTRY"
 fi
 
-test_failed=
+export e2e_reports_dir="$REPORTSDIR"
+if [ ! -d "$e2e_reports_dir" ] ; then
+    echo "Reports directory $e2e_reports_dir does not exist"
+    exit 1
+fi
+
+test_failed=0
+
+# Run go test in directory specified as $1 (relative path)
+function runGoTest {
+    cd "$TESTDIR"
+    echo "Running go test in $PWD/\"$1\""
+    if [ -z "$1" ] || [ ! -d "$1" ]; then
+        return 1
+    fi
+
+    cd "$1"
+    if ! go test -v . -ginkgo.v -ginkgo.progress -timeout 0; then
+        return 1
+    fi
 
+    return 0
+}
+
+# Check if $2 is in $1
+contains() {
+    [[ $1 =~ (^|[[:space:]])$2($|[[:space:]]) ]] && return 0  || return 1
+}
+
+echo "list of tests: $TESTS"
 for dir in $TESTS; do
-  cd "$SCRIPTDIR/../test/e2e/$dir"
-  if ! go test -v . -ginkgo.v -ginkgo.progress -timeout 0 ; then
-    test_failed=1
-    break
-  fi
-  if ! ("$SCRIPTDIR"/e2e_check_pod_restarts.sh) ; then
-      test_failed=1
-      break
+  # defer uninstall till after other tests have been run.
+  if [ "$dir" != "uninstall" ] ;  then
+      if ! runGoTest "$dir" ; then
+          test_failed=1
+          break
+      fi
+
+      if ! ("$SCRIPTDIR"/e2e_check_pod_restarts.sh) ; then
+          test_failed=1
+          if [ "$ON_FAIL" == "stop" ]; then
```

Comment:
It looks like ON_FAIL only applies to the re-start check? Maybe state this in the help output.